monkey/infection_monkey/island_api_client/http_island_api_client.py

```diff
@@ -45,16 +45,33 @@ class HTTPIslandAPIClient(IIslandAPIClient):
     """
 
     def __init__(
-        self, agent_event_serializer_registry: AgentEventSerializerRegistry, http_client: HTTPClient
+        self,
+        agent_event_serializer_registry: AgentEventSerializerRegistry,
+        http_client: HTTPClient,
+        otp: str,
     ):
         self._agent_event_serializer_registry = agent_event_serializer_registry
         self.http_client = http_client
+        self._otp = otp
 
     def connect(
         self,
         island_server: SocketAddress,
     ):
-        self.http_client.connect(island_server)
+        try:
+            self.http_client.set_server(island_server)
+            self.http_client.get("?action=is-up")
```

Comment:
Could we bury this `get()` down inside `set_server()`? In other words, can we define `set_server()` as successful if and only if it's able to `get("?action=is-up")`? This would reduce temporal coupling and error handling in this method. We wouldn't need to reset server to `None`.

Reply:
If we did this, `connect()` may become a more reasonable name once again.

Reply:
If we're concerned about abstraction layers, perhaps we should define it as `connect(server, connection_endpoint)` and call it with `connect(island_server, "?action=is-up")`. This keeps the Island specifics out of the `HTTPClient` but still reduces temporal coupling and simplifies `HTTPIslandAPIClient.connect()`.

Reply:
There's no `connect` method in HTTP. `HTTPClient` has temporal coupling because `HTTPIslandAPIClient` has it. This indicates an improper separation of concerns between the two. I've solved it by building the `HTTPClient` in `HTTPIslandAPIClient`.

Reply:
Let's finish this discussion in [this](https://github.com/guardicore/monkey/pull/3111#discussion_r1135946191) thread, since it's basically the same discussion.